### PR TITLE
pepperball gun

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
makes pepperball gun
it makes drain plasma gas on hit
it uses bimmers water gun sprite
marines get 1 in req vendors for testing

## Why It's Good For The Game
adds guns for cool more options, its cool

## Changelog
:cl:
add: Added new pepperball gun for marines
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -315. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

File: code/modules/projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -1136,3 +1136,31 @@ obj/item/weapon/gun/minigun/Fire(atom/target, mob/living/user, params, reflex =
...
 	else
 		return in_chamber ? (current_mag.current_rounds + 1) : current_mag.current_rounds
+
+// pepperball gun
+
+//-------------------------------------------------------
+//T-29 Smart Machine Gun (It's more of a rifle than the SG.)
+
+/obj/item/weapon/gun/rifle/pepperball
+	name = "\improper Pepperball gun"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/specialist.dm:1146]:
```suggestion
	name = "\improper TLNL-12 pepperball gun"
```

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -1136,3 +1136,31 @@ obj/item/weapon/gun/minigun/Fire(atom/target, mob/living/user, params, reflex =
...
+	desc = "A pepperball gun that fires toxins. It fires pepperball balls."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/specialist.dm:1147]:
```suggestion
	desc = "A TerraGov less-lethal weapon, it comes in limited numbers and has limited use throughout the military, though sizeable use throughout TerraGov's Navy Police. The variant fires plasma-draining tanglefoot pepperballs in a compressed form, quickly delivering ammunition throughout the field.."
```

File: code/modules/projectiles/guns/specialist.dm [hunk 3]
```diff
@@ -1136,3 +1136,31 @@ obj/item/weapon/gun/minigun/Fire(atom/target, mob/living/user, params, reflex =
...
+	icon_state = "pepperball"
+	item_state = "pepperball"
+	caliber = "pepperball" //codex
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/specialist.dm:1150]:
```suggestion
	caliber = "TGF-Ball" //codex
```

File: code/modules/projectiles/magazines/specialist.dm [hunk 4]
```diff
@@ -278,3 +278,14 @@ obj/item/ammo_magazine/rifle/m4ra/smart
 	w_class = WEIGHT_CLASS_NORMAL
 	gun_type = /obj/item/weapon/gun/minigun
 
+// pepperball
+/obj/item/ammo_magazine/rifle/pepperball
+	name = "\improper Pepperball magazine (10x28mm)"
+	desc = "A magazine of pepperball ammo. Fires plasma draining toxins."
+	caliber = "pepperball"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:285]:
```suggestion
	caliber = "TGF-Ball"
```

**Comment by TiviPlus** [comment]
MJP there are no sprites

**Comment by Pariah919** [comment]
image

**Comment by BraveMole** [comment]
sprite is awfull, please git ban

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -1615,6 +1615,45 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	name = "pepperball"
+	icon_state = "bullet"
+
+	flags_ammo_behavior = AMMO_BALLISTIC
+	sound_hit 	 = "ballistic_hit"
+	sound_armor  = "ballistic_armor"
+	sound_miss	 = "ballistic_miss"
+	sound_bounce = "ballistic_bounce"
+	var/datum/effect_system/smoke_spread/smoke_system
+	var/danger_message = "<span class='danger'>The pepperball explodes with a splat and explodes into purple gas!</span>"
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1628]: Declare new vars after redefining old ones

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

File: code/modules/projectiles/guns/specialist.dm [hunk 6]
```diff
@@ -1136,3 +1136,32 @@ obj/item/weapon/gun/minigun/Fire(atom/target, mob/living/user, params, reflex =
...
 		return in_chamber ? (current_mag.current_rounds + 1) : current_mag.current_rounds
+
+// pepperball gun
+
+//-------------------------------------------------------
+// its a peppeball gun
+
+/obj/item/weapon/gun/rifle/pepperball
+	name = "\improper TLLL-12 pepperball gun"
+	desc = "A TerraGov less-lethal weapon, it comes in limited numbers and has limited use throughout the military, though sizeable use throughout TerraGov's Navy Police. The variant fires plasma-draining tanglefoot pepperballs in a compressed form, quickly delivering ammunition throughout the field.."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/specialist.dm:1147]:
```suggestion
	desc = "A TerraGov less-lethal weapon, it comes in limited numbers and has limited use throughout the military, though sizeable use throughout TerraGov's Navy Police. The variant fires plasma-draining tanglefoot pepperballs in a compressed form, quickly delivering ammunition throughout the field."
```

File: code/modules/projectiles/magazines/specialist.dm [hunk 7]
```diff
@@ -278,3 +278,14 @@ obj/item/ammo_magazine/rifle/m4ra/smart
 	w_class = WEIGHT_CLASS_NORMAL
 	gun_type = /obj/item/weapon/gun/minigun
 
+// pepperball
+/obj/item/ammo_magazine/rifle/pepperball
+	name = "\improper Pepperball magazine (10x28mm)"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:283]:
```suggestion
	name = "\improper TGF-Pepperball magazine (TGF-Ball)"
```

**Comment by Pariah919** [comment]
test has been done

Idea no good, back to the drawing board

**Pull request closed without merging**